src/code/m_actor.c

```diff
@@ -0,0 +1,1143 @@
+
+    for (part = 0; part < ACTOR_PART_MAX; part++) {
+        Actor* actor;
+
+        if (restore_flag[part] == 1) {
+            for (actor = actorInfo->actorLists[part].head; actor != NULL; actor = actor->next) {
+                restore_fgdata(actor, play);
+            }
+        } else {
+            for (actor = actorInfo->actorLists[part].head; actor != NULL; actor = actor->next) {
+                if (actor->unk_003 == 1) {
+                    restore_fgdata(actor, play);
+                }
+            }
+        }
+    }
+}
+
+void Actor_info_save_actor(PlayState* play) {
+    ActorInfo* actorInfo = &play->actorInfo;
+    ActorPart part;
+
+    for (part = 0; part < ACTOR_PART_MAX; part++) {
+        Actor* actor;
+
+        for (actor = actorInfo->actorLists[part].head; actor != NULL; actor = actor->next) {
+            if (actor->save != NULL) {
+                actor->save(actor, play);
+                actor->save = NULL;
+            }
+        }
+    }
+
+    restore_fgdata_all(play);
+}
+
+Actor* Actor_info_delete(ActorInfo* actorInfo, Actor* actor, PlayState* play) {
+    Actor* newHead;
+    s32 pad UNUSED;
+    ActorOverlay* overlayEntry;
+    s32 fgName = actor->fgName;
+
+    overlayEntry = actor->overlayEntry;
+
+    restore_fgdata_one(actor, play);
+    Actor_dt(actor, play);
+
+    newHead = Actor_info_part_delete(actorInfo, actor);
+
+    switch (ACTOR_FGNAME_GET_F000(fgName)) {
+        case FGNAME_F000_D:
+        case FGNAME_F000_E:
+            common_data.unk_1004C->unk_10(actor);
+            break;
+
+        case FGNAME_F000_5:
+            common_data.unk_10098->unk_10(actor);
+            break;
+
+        default:
+            zelda_free(actor);
+            break;
+    }
+
+    if (overlayEntry->vramStart != NULL) {
+        overlayEntry->numLoaded--;
+        actor_free_check(overlayEntry, actor->fgName);
+    }
+
+    return newHead;
+}
+
+Actor* Actor_info_name_search_sub(Actor* actor, s16 name) {
+    for (; actor != NULL; actor = actor->next) {
+        if (name == actor->name) {
+            break;
+        }
+    }
+    return actor;
+}
+
+Actor* Actor_info_name_search(ActorInfo* actorInfo, s16 name, ActorPart part) {
+    Actor* head = actorInfo->actorLists[part].head;
+
+    if (head != NULL) {
+        return Actor_info_name_search_sub(head, name);
+    }
+    return NULL;
+}
+
+Actor* Actor_info_fgName_search_sub(Actor* actor, u16 fgName) {
+    for (; actor != NULL; actor = actor->next) {
+        if (actor->fgName == fgName) {
+            break;
+        }
+    }
+
+    return actor;
+}
+
+Actor* Actor_info_fgName_search(ActorInfo* actorInfo, u16 fgName, ActorPart part) {
+    Actor* head = actorInfo->actorLists[part].head;
+
+    if (head != NULL) {
+        return Actor_info_fgName_search_sub(head, fgName);
+    }
+    return NULL;
+}
+
+// unused
+void Part_Break_init(Part_Break* partBreak, s32 count, s32 arg2 UNUSED) {
+    size_t size1;
+    size_t size2;
+    size_t size3;
```

Comment:
this corresponds to `BodyBreak_Alloc` in z64 (neat that the term body break was close)
you can use that to name things if you want, since its unused and this game prob wont give you any more context

Reply:
guess its oot only actually? body break was refactored out in mm unless im mistaken.
are the gotos in the function required? if so, then oot will prob need to change for retail

Reply:
Oh I forgot to look at OoT. I looked at MM and since it wasn't there I thought this was some weird new stuff (but I remembered seeing a function like this somewhere).

And no, the gotos aren't required, I was curious if this function could be rewritten using gotos because it seems somewhat cleaner this way, and forgot to undo it :sweat_smile: . Either way this pattern is common for C programming, even I have done it myself, so this may have been the way it was originally written, but there's no proof and I can understand if we prefer to rewrite it without the gotos

Reply:
I think the way oot has it is very sensible and doesnt really need the gotos. This was merged already, but yeah something to consider for later